lib/reek/smells/subclassed_from_core_class.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+require_relative 'smell_detector'
+require_relative 'smell_warning'
+
+module Reek
+  module Smells
+    # TODO: documentation
+    class SubclassedFromCoreClass < SmellDetector
+      CORE_CLASSES = [:Array, :Hash, :String]
+
+      def self.contexts
+        [:class]
+      end
+
+      # Checks +ctx+ for if it is subclasssed from a core class
+      #
+      # @return [Array<SmellWarning>]
+      def inspect(ctx)
+        _class_node, ancestor_node, _body_node = ctx.exp.to_a
+        return [] if ancestor_node.nil?
```

Comment:
The modules under `SexpExtensions` are included in the classes representing the AST nodes. The modules that is included depends on the node type.

So nodes of type `:class` get `ClassNode` included, which is defined in `lib/reek/ast/sexp_extensions/module.rb`. In fact, you can see there that the relevant methods are defined in `ModuleNodeBase`, because many methods that are relevant for `:class` are also relevant for `:module` and `:casgn` nodes.

Long story short, you would probably want to define `ancestor` on `ClassNode` to start with, and then change the code above to:

```ruby
  exp = ctx.exp
  return [] unless exp.ancestor
```
```